### PR TITLE
Issues are sorted for all formatters

### PR DESCRIPTION
**Pull request by bastien-phi**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes

I use a lot phpinsights and we keep a `phpinsights.txt` file inside our project repositories (basically just the output of `php artisan insights --no-interaction -v`). It helps us to see which issues are solved or created when we develop features.

Unfortunately, diffs are sometimes very hard to read because order in insights details can change between platforms (linux or macos) or for any other obscure reason.

Moreover, the fact that details are not sorted by file makes more difficult find the issue and fix then in batch. For example, a sample of what I got : 
```
app/App/Admin/Filters/ByRuleTemplate.php:25: Unused parameter $request.
app/App/Dashboard/Controllers/UserController.php:54: Unused parameter $user
app/Support/Rules/HexColor.php:9: Unused parameter $attribute.
app/Support/Rules/ActivitySvg.php:17: Unused parameter $attribute.
app/Support/Rules/DomainName.php:17: Unused parameter $attribute.
app/Domain/User/Policies/UserPolicy.php:14: Unused parameter $admin.
```

This PR fixes this, sorting details of each insights, firstly by file, then line, function and finally message.

### CODE REVIEW

File: src/Domain/DetailsComparator.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Domain;
+
+final class DetailsComparator
```
Comment (olivernybroe) [inline src/Domain/DetailsComparator.php]: Add a phpdoc marking this as internal.

Comment (bastien-phi) [reply src/Domain/DetailsComparator.php]: done !

**Comment by bastien-phi** [comment]
@Jibbarth Should I add more test cases ?

**Comment by olivernybroe** [comment]
@bastien-phi Really nice! Could you add a test for the comparison order described in the PR, after that I'll say it is ready for merging 👍

**Comment by bastien-phi** [comment]
@olivernybroe is ee87ed0 what you wanted ?

**Comment by olivernybroe** [comment]
@bastien-phi Yep, it was something like that I had in mind, thanks a lot! ❤️

**Pull request merged**